src/parser/ast/mod.rs

```diff
@@ -11,32 +11,81 @@ pub mod tl;
-    /// Ending line number in the file
-    pub line_end: usize,
-    /// Ending column number in the file
-    pub col_end: usize,
+    /// Starting byte number in the file
+    pub start: usize,
+    /// Ending byte number in the file
+    pub end: usize,
     /// Source file reference
-    file: Rc<SimpleFile<String, String>>,
+    file: Arc<SimpleFile<String, String>>,
 }
 
 impl DebugSymRef {
-    pub fn new(
-        line_start: usize,
-        col_start: usize,
-        line_end: usize,
-        col_end: usize,
-        file: Rc<SimpleFile<String, String>>,
-    ) -> DebugSymRef {
-        DebugSymRef {
-            line_start,
-            col_start,
-            line_end,
-            col_end,
-            file,
+    pub fn new(start: usize, end: usize, file: Arc<SimpleFile<String, String>>) -> DebugSymRef {
+        DebugSymRef { start, end, file }
+    }
+
+    fn get_column_number(&self, line_index: usize, start: usize) -> usize {
+        match self.file.column_number((), line_index, start) {
+            Ok(number) => number,
+            Err(err) => {
+                panic!("Column number at {} not found: {}", line_index, err);
+            }
+        }
+    }
+
+    fn get_line_index(&self, start: usize) -> usize {
+        match self.file.line_index((), start) {
+            Ok(index) => index,
+            Err(err) => {
+                panic!("Line index at {} not found: {}", start, err);
+            }
+        }
+    }
+
+    fn get_line_number(&self, line_index: usize) -> usize {
+        match self.file.line_number((), line_index) {
+            Ok(number) => number,
+            Err(err) => {
+                panic!("Line number at {} not found: {}", line_index, err);
+            }
+        }
+    }
+
+    fn get_line_start(&self) -> usize {
+        let line_idx = self.get_line_index(self.start);
+        self.get_line_number(line_idx)
+    }
+
+    fn get_col_start(&self) -> usize {
+        let line_idx = self.get_line_index(self.start);
+        self.get_column_number(line_idx, self.start)
+    }
+
+    fn get_line_end(&self) -> usize {
+        let line_idx = self.get_line_index(self.end);
+        self.get_line_number(line_idx)
+    }
+
+    fn get_col_end(&self) -> usize {
+        let line_idx = self.get_line_index(self.end);
+        self.get_column_number(line_idx, self.end)
+    }
+
+    pub(crate) fn get_filename(&self) -> String {
+        self.file.name().to_string()
+    }
+
+    /// Returns the proximity score of the given offset to the debug symbol.
+    /// The proximity score is the sum of the distance from the start and end of the symbol.
+    /// If the offset is not within the symbol, -1 is returned.
+    pub fn proximity_score(&self, offset: usize) -> i32 {
+        if self.start <= offset && offset <= self.end {
+            let start_diff = offset as i32 - self.start as i32;
+            let end_diff = self.end as i32 - offset as i32;
+
+            start_diff + end_diff
```

Comment:
I think it would be more idiomatic to return Option<u32> Instead of -1

Reply:
Could you add tests for this method

Reply:
I think if `self.start <= offset <= self.end` then `offset - self.start + self.end - offset == self.start + self.end`. I think the value that you want is `self.end - self.start`. And for that you don't need offset. It seems what you want is the size of the symbol footprint.